pkg/config/migration/four-to-five.go

```diff
@@ -0,0 +1,246 @@
+      var provisionDefaultsVar = "GSB_SERVICE_GOOGLE_"+svcName+"_PROVISION_DEFAULTS";
+
+			if (! anyExist([customPlanVar, enabledVar, bindDefaultsVar, provisionDefaultsVar])) {
+				return null;
+			}
+
+			// explicitly check for boolean enabled false and turn it into a string.
+			var enabled = lookupProp(enabledVar);
+			if (enabled === false) {
+				enabled = 'false';
+			}
+
+      var context = {
+        "custom_plans": lookupProp(customPlanVar) || '[]',
+        "enabled": enabled || 'true',
+        "bind_defaults": lookupProp(bindDefaultsVar) || '{}',
+        "provision_defaults": lookupProp(provisionDefaultsVar) || '{}',
+      }
+
+      deleteProp(customPlanVar);
+      deleteProp(enabledVar);
+      deleteProp(bindDefaultsVar);
+      deleteProp(provisionDefaultsVar);
+
+      // convert to plain object
+      for (var key in context) {
+        context[key] = JSON.parse(context[key]);
+      }
+
+			context["disabled"] = !context["enabled"];
+			delete context["enabled"];
+
+      context["//"] = "Builtin " + svcName;
+      return context;
+    }
+
+    function mergeServices(envVar) {
+      var out = {
+        "f80c0a3e-bd4d-4809-a900-b4e33a6450f1": mergeService("BIGQUERY"),
+        "b8e19880-ac58-42ef-b033-f7cd9c94d1fe": mergeService("BIGTABLE"),
+        "4bc59b9a-8520-409f-85da-1c7552315863": mergeService("CLOUDSQL_MYSQL"),
+        "cbad6d78-a73c-432d-b8ff-b219a17a803a": mergeService("CLOUDSQL_POSTGRES"),
+        "3e897eb3-9062-4966-bd4f-85bda0f73b3d": mergeService("DATAFLOW"),
+        "76d4abb2-fee7-4c8f-aee1-bcea2837f02b": mergeService("DATASTORE"),
+        "e84b69db-3de9-4688-8f5c-26b9d5b1f129": mergeService("DIALOGFLOW"),
+        "a2b7b873-1e34-4530-8a42-902ff7d66b43": mergeService("FIRESTORE"),
+        "5ad2dce0-51f7-4ede-8b46-293d6df1e8d4": mergeService("ML_APIS"),
+        "628629e3-79f5-4255-b981-d14c6c7856be": mergeService("PUBSUB"),
+        "51b3e27e-d323-49ce-8c5f-1211e6409e82": mergeService("SPANNER"),
+        "83837945-1547-41e0-b661-ea31d76eed11": mergeService("STACKDRIVER_DEBUGGER"),
+        "2bc0d9ed-3f68-4056-b842-4a85cfbc727f": mergeService("STACKDRIVER_MONITORING"),
+        "00b9ca4a-7cd6-406a-a5b7-2f43f41ade75": mergeService("STACKDRIVER_PROFILER"),
+        "c5ddfe15-24d9-47f8-8ffe-f6b7daa9cf4a": mergeService("STACKDRIVER_TRACE"),
+        "b9e4332e-b42b-4680-bda5-ea1506797474": mergeService("STORAGE")
+      };
+
+			Object.keys(out).forEach(function(key) {
+			if (out[key] == null) delete out[key];
+		});
+
+
+			for (var key in out) {
+				if(out[key] == null) {
+					delete out[key];
+				}
+			}
+
+      setProp(envVar, JSON.stringify(out, null, "  "));
+    }`,
+		TransformFuncName: "mergeServices",
+	}
+
+	return Migration{
+		Name:       "Merge Service Config",
+		TileScript: jst.ToJs(),
+		GoFunc:     jst.RunGo,
+	}
+}
+
+// DeleteWhitelistKeys removes the whitelist keys used prior to 4.0
+func DeleteWhitelistKeys() Migration {
+	whitelistKeys := []string{
```

Comment:
Any reason not to just in-line this with the function call?

Reply:
Done